migrations/20190520221710-no-paranoid.js

```diff
@@ -0,0 +1,300 @@
+/**
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * migrations/20190520221710-no-paranoid.js
+ *
+ * Note: the *new* indices will get created on db startup *without* any
+ * migration code, which means they cannot be created as *unique* until
+ * *after* the "destroySoftDeleted" step has completed!
+ */
+'use strict';
+const Promise = require('bluebird');
+const db = require('../db/index');
+const IS = 'isDeleted';
+const AT = 'deletedAt';
+
+/**
+ * If a collector references a soft-deleted collector group, clear that
+ * collector group from the collector.
```

Comment:
Do we need to do this from the generator side as well?
Are there any other associations we need to do this for?

Reply:
These were the only ones where deletion was blocked for referential integrity in either sandbox or prod, I tried to think through other scenarios where similar things could happen but couldn't come up with any.